packages/core/src/common/message-service.ts

```diff
@@ -25,35 +25,114 @@ import {
 } from './message-service-protocol';
 import { CancellationTokenSource } from './cancellation';
 
+/**
+ * Service to log and categorize messages, show progress information and offer actions.
+ *
+ * The messages are processed by this service and forwarded to an injected {@link MessageClient}.
+ * Usually this is the {@link NotificationManager} contributed by "@theia/messages" rendering
+ * messages as notifications in the frontend.
```

Comment:
Do you believe we should reference `@theia/messages` (a downstream extension) in the parent `@theia/core`?

Reply:
I just wanted to give an example of a `MessageClient`. I assume that many apps will use the `NotificationManager` so when someone looks at what the `MessageService` does they know that their messages end up there. So personally I think it's fine to list it as an example here, but if you prefer I could also remove it again.

Reply:
I removed the link and reworded it a bit. If you prefer to remove the mention of "@theia/messages" completely, just let me know :)